### ISSUE TITLE
Attempted to call an undefined method named "checkSecurity"

### ISSUE THREAD

**Issue by benschumi** [twigphp/Twig#3411]
Hello, 

it's seem what something is broken when i updated in twig 1.44

<pre>Attempted to call an undefined method named "checkSecurity" of class "__TwigTemplate_ab8c5b3634ec2ea857069474b49ae5f5b2f484499df40afbe2063e28cae810c4".
500 Internal Server Error - UndefinedMethodException

Stack Trace
in /var/www/vhosts/vendor/twig/twig/src/Extension/CoreExtension.php at line 1572  -
    // if a Template instance is passed, it might have been instantiated outside of a sandbox, check security
    if ($template instanceof TemplateWrapper || $template instanceof Template) {
        $template->unwrap()->checkSecurity();
    }
    $loaded = null;
</pre>

happen with a {{ include(template_from_string("something")) }}

**Comment by benschumi** [comment]
it's work with a {% include template_from_string("something") %} but not with {{ include(template_from_string("something")) }}

In the doc, it's writtent with {{ include() }}
https://twig.symfony.com/doc/1.x/functions/template_from_string.html

**Comment by stof** [comment]
Are you using the SandboxExtension or no ?

**Comment by stof** [comment]
seems to be caused by #3349 which does not account for templates compiled without the sandbox and so without a `checkSecurity` method.

**Comment by benschumi** [comment]
We don't use the sandbox

**Comment by stempora** [comment]
After upgrading from 3.0.5 to 3.1.0 i received the same error:

Call to undefined method __TwigTemplate_ffcada2905c270c1bb8149ef10195d5fdfc3992f37169ff72fca2cffa2148cdf::checkSecurity()
File: /twig/twig/src/Extension/CoreExtension.php
Line: 1250

**Comment by stempora** [comment]
After adding the sandbox extension it started to work again.

**Comment by ruudk** [comment]
Same issue here, we are not using the sandbox extension and upgrading to 3.1.0 fails on `{{ include(template_from_string(var)) }}`

**Comment by klipach** [comment]
Same issue after upgrading from 2.13.1 to 2.14.0

**Comment by fabiomosti** [comment]
experiencing the same issue from version 2.14.x and higher

**Comment by ruudk** [comment]
@Warxcell Could you please have a look?

**Comment by abgit** [comment]
same issue here

**Comment by ThomasLandauer** [comment]
I "fixed" it by adding this to my `composer.json`, then run `composer update`
```json
    "conflict": {
        "symfony/symfony": "*",
        "twig/twig": "2.14.0"
    },
```

**Comment by gnat42** [comment]
Yeah I've hit the same issue on two projects. 2.14.0 (reverting to 2.13.1 fixes it).
In our case I'm not using the template_from_string extension but have something that is manually rendering via 

```
$this->twig->createTemplate($data)->render($fullParams);
```

Which is essentially what the template_from_string extension does. I'm wondering in my case without a sandbox, how should we call the above? it is most definitely caused by #3349, but it isn't obvious to me how I should be creating the template to account for the sandbox or lack thereof...

**Comment by gnat42** [comment]
No idea if this is the appropriate fix but this would be correct:
CoreExtension.php around line 1216 (Twig 2.14.0)

```php
    // if a Template instance is passed, it might have been instantiated outside of a sandbox, check security
    if ($template instanceof TemplateWrapper || $template instanceof Template) {
        $tmp = $template->unwrap();
        if (method_exists($tmp, 'checkSecurity')) {
            $tmp->checkSecurity();
        }
    }
```

**Comment by curruwilla** [comment]
What is the solution? I have the same problem

**Comment by fabpot** [comment]
Can you check that #3413 fixes your issue?

**Comment by ruudk** [comment]
FWIW > I applied the fix to 3.1.0 and it solved the issue.
